stock_prices/stock_prices.py

```diff
@@ -3,7 +3,37 @@
 import argparse
 
 def find_max_profit(prices):
-  pass
+  low = prices[0]
+  i = 0
+  while i < len(prices):
+    if prices[i] < low:
+      low = prices[i]
+    i += 1
+  if low == prices[-1]:
+    k = 0
+    secondLow = prices[0]
+    while k < len(prices) - 1:
+      if prices[k] < secondLow:
+        secondLow = prices[k]
+      k += 1
+    j = prices.index(secondLow)
+    high = prices[prices.index(secondLow) + 1]
+    while j < len(prices):
+      if prices[j] > high:
+        high = prices[j] 
+      j += 1
+    if high == secondLow:
+      return -(high - low)
+    else:
+      return high - secondLow
+  else:
+    j = prices.index(low)
+    high = prices[prices.index(low) + 1]
+    while j < len(prices):
+      if prices[j] > high:
+        high = prices[j] 
+      j += 1
+    return high - low
```

Comment:
Hey Alberto, you've got yourself a working solution! 🎉 While it will get the job done, it is quite expensive to run. With some crafty variable declaration before your starts, you should be able to reduce the size of your function to either a `while` statement or a couple of `if/else` statements.